tests/api/auth/backend/test_basic_auth.py

```diff
@@ -0,0 +1,135 @@
+        tester = self.appbuilder.sm.find_user(username="test")
+        if not tester:
+            self.appbuilder.sm.add_user(
+                username="test",
+                first_name="test",
+                last_name="test",
+                email="test@fab.org",
+                role=role_admin,
+                password="test",
+            )
+
+    def test_success(self):
+        token = "Basic " + b64encode(b"test:test").decode()
+        clear_db_pools()
+
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert current_user.email == "test@fab.org"
+
+        assert response.status_code == 200
+        assert response.json == {
+            "pools": [
+                {
+                    "name": "default_pool",
+                    "slots": 128,
+                    "occupied_slots": 0,
+                    "running_slots": 0,
+                    "queued_slots": 0,
+                    "open_slots": 128,
+                },
+            ],
+            "total_entries": 1,
+        }
+
+    @parameterized.expand([
+        ("basic",),
+        ("basic ",),
+        ("bearer",),
+        ("test:test",),
+        (b64encode(b"test:test").decode(),),
+        ("bearer ",),
+        ("basic: ",),
+        ("basic 123",),
+    ])
+    def test_malformed_headers(self, token):
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert response.status_code == 401
+            assert response.headers["Content-Type"] == "application/problem+json"
+            assert response.headers["WWW-Authenticate"] == "Basic"
+            assert response.json == {
+                'detail': None,
+                'status': 401,
+                'title': 'Unauthorized',
+                'type': 'about:blank',
+            }
+
+    @parameterized.expand([
+        ("basic " + b64encode(b"test").decode(),),
+        ("basic " + b64encode(b"test:").decode(),),
+        ("basic " + b64encode(b"test:123").decode(),),
+        ("basic " + b64encode(b"test test").decode(),),
+    ])
+    def test_invalid_auth_header(self, token):
+        with self.app.test_client() as test_client:
+            response = test_client.get(
+                "/api/v1/pools", headers={"Authorization": token}
+            )
+            assert response.status_code == 401
+            assert response.headers["Content-Type"] == "application/problem+json"
+            assert response.headers["WWW-Authenticate"] == "Basic"
+            assert response.json == {
+                'detail': None,
+                'status': 401,
+                'title': 'Unauthorized',
+                'type': 'about:blank',
+            }
+
+    def test_experimental_api(self):
```

Comment:
@mik-laj packed experimental api related checks into this single test so it's easy to delete in the future.